File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        int numOfOpened=0;
        for(int idx=0;idx<s.length();idx++){
            switch(s.charAt(idx)){
                case '(': {
                    numOfOpened++;
                    break;
                }
                case ')': {
                    numOfOpened--;
                    if(numOfOpened<0) return false;
                    break;
                }
                default: throw new RuntimeException();
            }
        }
        return numOfOpened==0;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let numOfOpened=0;
    for(const char of s.split("")){
        switch(char){
            case "(":
                numOfOpened++;
                break;
            case ")":
                numOfOpened--;
                if(numOfOpened<0){
                    return false;
                }
        }
    }
    return numOfOpened===0;
}